service/history/workflow/retry_test.go

```diff
@@ -175,210 +174,220 @@ func Test_NextRetry(t *testing.T) {
-
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(time.Millisecond*8, interval)
-	a.Equal(enumspb.RETRY_STATE_IN_PROGRESS, retryState)
-	ai.Attempt++
-
-	// no retry as max attempt reached
-	a.EqualValues(ai.RetryMaximumAttempts, ai.Attempt)
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(backoff.NoBackoff, interval)
-	a.Equal(enumspb.RETRY_STATE_MAXIMUM_ATTEMPTS_REACHED, retryState)
-
-	// increase max attempts, with max interval cap at 10s
-	ai.RetryMaximumAttempts = 6
-	ai.RetryMaximumInterval = durationpb.New(10 * time.Millisecond)
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(time.Millisecond*10, interval)
-	a.Equal(enumspb.RETRY_STATE_IN_PROGRESS, retryState)
-	ai.Attempt++
-
-	// no retry because expiration time before next interval
-	ai.RetryMaximumAttempts = 8
-	ai.RetryExpirationTime = timestamppb.New(now.Add(time.Millisecond * 5))
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(backoff.NoBackoff, interval)
-	a.Equal(enumspb.RETRY_STATE_TIMEOUT, retryState)
-
-	// extend expiration, next interval should be 10s
-	ai.RetryExpirationTime = timestamppb.New(now.Add(time.Minute))
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(time.Millisecond*10, interval)
-	a.Equal(enumspb.RETRY_STATE_IN_PROGRESS, retryState)
-	ai.Attempt++
-
-	// with big max retry, math.Pow() could overflow, verify that it uses the MaxInterval
-	ai.Attempt = 64
-	ai.RetryMaximumAttempts = 100
-	interval, retryState = getBackoffInterval(
-		now,
-		ai.Attempt,
-		ai.RetryMaximumAttempts,
-		ai.RetryInitialInterval,
-		ai.RetryMaximumInterval,
-		ai.RetryExpirationTime,
-		ai.RetryBackoffCoefficient,
-		serverFailure,
-		ai.RetryNonRetryableErrorTypes,
-	)
-	a.Equal(time.Millisecond*10, interval)
-	a.Equal(enumspb.RETRY_STATE_IN_PROGRESS, retryState)
-	ai.Attempt++
+	t.Run("should not retry if there is no retry policy", func(t *testing.T) {
```

Comment:
This is great! Every time when there is an opportunity to use error message (or test name, like here) it should be used instead of comment.